ns2-stat-api/src/main.rs

```diff
@@ -34,49 +33,65 @@ struct DatedData<T> {
     data: T,
 }
 
+trait Dated<T: Ord> {
+    fn date(&self) -> T;
+}
+
+impl Dated<u32> for GameStats {
+    fn date(&self) -> u32 {
+        self.round_info.round_date
+    }
+}
+
+impl Dated<u32> for &NS2Stats {
```

Comment:
```suggestion
impl Dated<u32> for NS2Stats {
```

Reply:
This sadly doesn't work since we can't move out of an `Arc`, but we can reference it

Reply:
IMO the better solution here would be to add an `impl<T: Dated<u32>> Dated<u32> for &T` then.